src/main/java/usersandpostsCRUD/demo/controller/PostController.java

```diff
@@ -0,0 +1,42 @@
+package usersandpostsCRUD.demo.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import usersandpostsCRUD.demo.dto.PostRequestDto;
+import usersandpostsCRUD.demo.dto.UserDto;
+import usersandpostsCRUD.demo.entity.Post;
+import usersandpostsCRUD.demo.service.PostService;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/posts")
+public class PostController {
+
+    @Autowired
+    private PostService postService;
+
+    @GetMapping
+    public List<Post> getAllPosts() {
+        return postService.getAllPosts();
+    }
+
+    @PostMapping
+    public Post createPost(@RequestBody PostRequestDto postRequest) {
+        Post post = new Post();
+        post.setTitle(postRequest.getTitle());
+        post.setDescription(postRequest.getDescription());
+        return postService.createPost(post, postRequest.getUserId());
```

Comment:
Why sending post and postRequest.getUserId ?
You shoud create the Post object in the service.